migrations/20220905173159__rubrics__create.sql

```diff
@@ -0,0 +1,58 @@
+CREATE TABLE IF NOT EXISTS rubrics (
+    id BIGSERIAL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    starting_points DOUBLE PRECISION NOT NULL,
+    max_points DOUBLE PRECISION NOT NULL,
+    min_points DOUBLE PRECISION NOT NULL,
+    modified_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    deleted_at TIMESTAMP WITH TIME ZONE
+);
+
+CREATE TABLE IF NOT EXISTS rubric_items (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_id BIGINT NOT NULL REFERENCES rubrics(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    number BIGINT NOT NULL,
+    points DOUBLE PRECISION NOT NULL,
+    short_text TEXT,
+    description TEXT,
+    staff_instructions TEXT,
+    key_binding TEXT,
+    deleted_at TIMESTAMP WITH TIME ZONE
+);
+
+CREATE INDEX IF NOT EXISTS rubric_items_rubric_id ON rubric_items(rubric_id);
+
+CREATE TABLE IF NOT EXISTS rubric_gradings (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_id BIGINT NOT NULL REFERENCES rubrics(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    computed_points DOUBLE PRECISION NOT NULL,
+    adjust_points DOUBLE PRECISION NOT NULL,
+    starting_points DOUBLE PRECISION NOT NULL,
+    max_points DOUBLE PRECISION NOT NULL,
+    min_points DOUBLE PRECISION NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS rubric_grading_items (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_grading_id BIGINT NOT NULL REFERENCES rubric_gradings(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    rubric_item_id BIGINT NOT NULL REFERENCES rubric_items(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    score DOUBLE PRECISION NOT NULL DEFAULT 1,
+    points DOUBLE PRECISION NOT NULL,
+    short_text TEXT,
+    note TEXT
+);
+
+-- TODO Consider other indices
```

Comment:
I'll remove this comment. I was considering cases where gradings are selected bottom up instead of top down, but given gradings can be associated to grading jobs and old logs, that would select rows that are no longer active, so top-down is the way to go.

Reply:
Removed in ad969bcb5.